db/model/collector.js

```diff
@@ -130,17 +131,16 @@ module.exports = function collector(seq, dataTypes) {
 
       afterUpdate(inst /* , opts */) {
         if (inst.changed('status')) {
-          /* if status is changed to Running, then find and assign unassigned
-           generators */
           if (inst.status === collectorStatus.Running) {
+            /* if status is changed to Running, then find and assign unassigned
+             generators */
             return u.assignUnassignedGenerators();
-          }
-
-          if (inst.status === collectorStatus.Stopped ||
-            inst.status === collectorStatus.Paused) {
-            /* if status is changed to Stopped or Paused, then reassign the
-             generators which were assigned to this collector */
-            return inst.reassignGenerators();
+          } else if (inst.previous('status') === collectorStatus.Running) {
```

Comment:
Is there a test for this use case = i.e. reassigning generator when previous status='Running'?

Reply:
This is logically equivalent to what it was before, I just simplified it. The tests you wrote in collectors/pause and collectors/stop cover this.